From 5c9e4050c1e8c66b1b8b4e93d48b8e23ff1ea818 Mon Sep 17 00:00:00 2001
From: Carlo Cabrera <30379873+carlocab@users.noreply.github.com>
Date: Wed, 6 Sep 2023 03:35:47 +0800
Subject: [PATCH] [Build] fix `gRPCConfig.cmake` (#33361)

Without this change, calling `find_package(gRPC REQUIRED)` errors out
with

CMake Error at /opt/homebrew/lib/cmake/grpc/gRPCConfig.cmake:8
(find_dependency):
      Unknown CMake command "find_dependency".

The issue is that `find_dependency` is provided by the
`CMakeFindDependencyMacro` module[^1], so we need to `include` it before
use.

[^1]:
https://cmake.org/cmake/help/v3.26/module/CMakeFindDependencyMacro.html
---
 cmake/protobuf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index 80181c6218..fab2eb00e9 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -86,6 +86,6 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
         set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
       endif()
     endif()
-    set(_gRPC_FIND_PROTOBUF "find_dependency(Protobuf CONFIG)")
+    set(_gRPC_FIND_PROTOBUF "include(CMakeFindDependencyMacro)\nfind_dependency(Protobuf CONFIG)")
   endif()
 endif()
-- 
2.42.0

